src/lkl/setup.c

```diff
@@ -1288,16 +1304,20 @@ static void* lkl_termination_thread(void* args)
     SGXLKL_VERBOSE("calling lkl_sys_halt()\n");
     res = lkl_sys_halt();
     if (res < 0)
-    {
         sgxlkl_fail("LKL halt, %s\n", lkl_strerror(res));
-    }
 
-    /* Notify host about the guest shutdown */
+    SGXLKL_VERBOSE("calling sgxlkl_host_shutdown_notification()\n");
     sgxlkl_host_shutdown_notification();
 
-    /* Set termination flag to notify lthread scheduler to bail out. */
-    lthread_notify_completion();
+    SGXLKL_VERBOSE("calling vio_terminate()\n");
+    vio_terminate();
```

Comment:
Should this go before the `lkl_sys_halt` call?  Or will VirtIO interrupts be silently ignored if they happen after the kernel has exited?  I vaguely remember some shutdown-gate code for this, but it would be good to have a comment explaining this assumption.

Reply:
If we move it earlier, there may still be background I/O by the kernel, no? (I tried this, and I saw failures.)